### PR TITLE
update limitrange example

### LINKED ISSUE #13718: Add an example which contains request constraint for LimitRange
Since resource QoS is implemented, limitrange also supports request constraint, we can add an example for that.

### PR DESCRIPTION
**Pull request by HaiyangDING**
Partially fix #13718 

I will update the README file accordingly sooner.

### CODE REVIEW

**Comment by HaiyangDING** [comment]
I can not see the information from Shippable, any ideas?

**Comment by HaiyangDING** [comment]
@derekwaynecarr 

I have changed the README file and the yaml file, please take a look.

File: docs/admin/limitrange/README.md [hunk 1]
```diff
@@ -86,30 +86,39 @@ Let's create a simple limit in our namespace.
...
-----      --------  ---  --- ---
-Pod       memory    6Mi  1Gi -
-Pod       cpu       250m   2 -
-Container memory    6Mi  1Gi 100Mi
-Container cpu       250m   2 250m
+Name:		mylimits
+Namespace:	limit-example
+Type		Resource	Min	Max	Request	Limit	Limit/Request
+----		--------	---	---	-------	-----	-------------
+Pod		cpu		200m	2	500m	1	-
```
Comment (derekwaynecarr) [inline docs/admin/limitrange/README.md]: request and limit are not supported for type Pod, just Container.

I will make a note to do a validation error if we encounter this.

File: docs/admin/limitrange/README.md [hunk 2]
```diff
@@ -86,30 +86,39 @@ Let's create a simple limit in our namespace.
...
 
 In this scenario, we have said the following:
 
-1. The total memory usage of a pod across all of its container must fall between 6Mi and 1Gi.
-2. The total cpu usage of a pod across all of its containers must fall between 250m and 2 cores.
-3. A container in a pod may consume between 6Mi and 1Gi of memory.  If the container does not
-specify an explicit resource limit, each container in a pod will get 100Mi of memory.
-4. A container in a pod may consume between 250m and 2 cores of cpu.  If the container does
-not specify an explicit resource limit, each container in a pod will get 250m of cpu.
+1. The total memory usage of a pod across all of its container(s) must fall between 6Mi and 1Gi.
```
Comment (derekwaynecarr) [inline docs/admin/limitrange/README.md]: see above comment that default request and default limit are not supported for pod limit type.

If a max constraint is specified for a resource, then a limit must be specified for that resource across all containers.  Failure to specify a limit will result in a validation error when attempting to create the pod.  

If a min constraint is specified for a resource, then a request must be specified for that resource across all containers.  Failure to specify a request will result in a validation error when attempting to create the pod.

So for this table entry, the sum of all containers memory requests must be >= 6Mi and the sum of all containers memory limits must be <= 1Gi.

File: docs/admin/limitrange/README.md [hunk 3]
```diff
@@ -86,30 +86,39 @@ Let's create a simple limit in our namespace.
...
+   If no *limit* and/or *request* are specified for memory, the pod will get 1Gi of memory *limit*
+   and/or 100Mi of memory *request*. This means that, by default, the pod is guaranteed to allocate
+   100Mi of memory and is able to consume up to 1Gi of memory if free resource allows. See more
+   details about *request* and *limit* in [Resource QoS](../../proposals/resource-qos.md).
+2. The total cpu usage of a pod across all of its container(s) must fall between 200m and 2 cores.
```
Comment (derekwaynecarr) [inline docs/admin/limitrange/README.md]: same basic comment on what min or max means as earlier.

File: docs/admin/limitrange/limits.yaml [hunk 4]
```diff
@@ -4,20 +4,29 @@ metadata:
   name: mylimits
 spec:
   limits:
-  - max:
+  - default:
```
Comment (derekwaynecarr) [inline docs/admin/limitrange/limits.yaml]: remove this.  i will open a PR to ensure that default and defaultRequest on a a Pod LimitType results in validation error.

**Comment by derekwaynecarr** [comment]
I think in the interest of completeness, we should note that CPU limits are not enforced in the default Kubernetes setup on the physical node that runs the container unless the administrator deploys the kubelet with the folllowing flag:

```
$ kubelet --help
Usage of kubelet
....
   --cpu-cfs-quota[=false]: Enable CPU CFS quota enforcement for containers that specify CPU limits
$ kubelet --cpu-cfs-quota=true ...
```

By default, the Debian distro I see used on GCE does not support CPU CFS quota flag and @vishh requested it default to false as a result.  This means on GCE at least, there is no limit to what you can burst on the CPU that is actually enforced yet in the kernel assuming unless there is contention on the CPU and shares kicks in.

I know on Red Hat distros, we will want this flag enabled since the kernel supports it out of the box.  In addition, it will be on by default in the OpenShift configurations.  With the flag on, even if there is excess CPU time available, your container will be throttled to not go over its allowed CPU period specified by your limit.  So for example, if you limit to 250m, on a Red Hat distro, you would be allowed to get up to 1/4 of the CPU time for every 100ms period on the node, but no more even if there is no contention on the CPU.

**Comment by derekwaynecarr** [comment]
Related issue to fix brought out by this example:
https://github.com/kubernetes/kubernetes/issues/13831

**Comment by HaiyangDING** [comment]
I was too busy in the past few days. I will update this PR sooner.

**Comment by HaiyangDING** [comment]
The PR is not done yet. I will update sooner. Sorry for the delayed update, being too busy these days.

**Comment by HaiyangDING** [comment]
@derekwaynecarr 

I have updated this PR, PTAL. Sorry for the delayed update.

**Comment by HaiyangDING** [comment]
@derekwaynecarr 

Hi, would you please take a look at this PR please? So that I could make further corrections if necessary. Thanks :)

**Comment by derekwaynecarr** [comment]
I will review tomorrow am - apologies for delay

File: docs/admin/limitrange/README.md [hunk 5]
```diff
@@ -94,22 +94,28 @@ Let's describe the limits that we have imposed in our namespace.
...
 $ kubectl describe limits mylimits --namespace=limit-example
 Name:   mylimits
-Type      Resource  Min  Max Default
-----      --------  ---  --- ---
-Pod       memory    6Mi  1Gi -
-Pod       cpu       250m   2 -
-Container memory    6Mi  1Gi 100Mi
-Container cpu       250m   2 250m
+Namespace:  limit-example
+Type    Resource  Min Max Request Limit Limit/Request
```
Comment (derekwaynecarr) [inline docs/admin/limitrange/README.md]: Align these column field values so it renders properly when viewed.

**Comment by derekwaynecarr** [comment]
the content LGTM, but there is one issue with alignment in tables.

fix and poke the PR so I can apply final LGTM.

**Comment by HaiyangDING** [comment]
@derekwaynecarr 

Hi, thanks for the review. The alignment has been fixed. PTAL.

**Comment by derekwaynecarr** [comment]
LGTM, thanks!

**Pull request merged**